File: Python/Film.py
class Film:
    # constructor
    def __init__(self, id:int, nama:str, durasi:int, rating:float):
        self.__id = id
        self.__nama = nama
        self.__durasi = durasi
        self.__rating = rating

    # Setter and Getter for ID
    def setId(self, id): self.__id = id
    def getId(self): return self.__id

    # Setter and Getter for nama
    def setNama(self, nama): self.__nama = nama
    def getNama(self): return self.__nama

    # Setter and Getter for durasi
    def setDurasi(self, durasi): self.__durasi = durasi
    def getDurasi(self): return self.__durasi

    # Setter and Getter for rating
    def setRating(self, rating): self.__rating = rating
    def getRating(self): return self.__rating

File: PHP/Film.php
<?php
class Film {
    private int $id;
    private string $nama;
    private int $durasi;
    private float $rating;
    private string $gambar;

    public function __construct(int $id = 0, string $nama = "", int $durasi = 0, float $rating = 0.0, string $gambar = "") {
        $this->id = $id;
        $this->nama = $nama;
        $this->durasi = $durasi;
        $this->rating = $rating;
        $this->gambar = $gambar;
    }

    // Getter dan Setter untuk id
    public function getId(): int { return $this->id; }
    public function setId(int $id): void { $this->id = $id; }

    // Getter dan Setter untuk nama
    public function getNama(): string { return $this->nama; }
    public function setNama(string $nama): void { $this->nama = $nama; }

    // Getter dan Setter untuk durasi
    public function getDurasi(): int { return $this->durasi; }
    public function setDurasi(int $durasi): void { $this->durasi = $durasi; }

    // Getter dan Setter untuk rating
    public function getRating(): float { return $this->rating; }
    public function setRating(float $rating): void { $this->rating = $rating; }

    // Getter dan Setter untuk gambar
    public function getGambar(): string { return $this->gambar; }
    public function setGambar(string $gambar): void { $this->gambar = $gambar; }
}
?>
